src/core/pointcloud/qgspointcloudstatistics.cpp

```diff
@@ -0,0 +1,128 @@
+ ***************************************************************************/
+
+#include "qgspointcloudstatistics.h"
+
+#include "qgspointcloudattribute.h"
+
+QgsPointCloudStatistics::QgsPointCloudStatistics()
+{
+
+}
+
+QgsPointCloudStatistics::QgsPointCloudStatistics( int sampledPointsCount, const QMap<QString, AttributeStatistics> &stats )
+  : mSampledPointsCount( sampledPointsCount ), mStatisticsMap( stats )
+{
+
+}
+
+void QgsPointCloudStatistics::clear()
+{
+  mStatisticsMap.clear();
+}
+
+void QgsPointCloudStatistics::clear( const QVector<QgsPointCloudAttribute> &attributes )
+{
+  for ( QgsPointCloudAttribute attribute : attributes )
+  {
+    mStatisticsMap.remove( attribute.name() );
+  }
+}
+
+QgsPointCloudStatistics::AttributeStatistics QgsPointCloudStatistics::statisticsOf( const QString &attribute ) const
+{
+  AttributeStatistics defaultVal;
+  defaultVal.minimum = std::numeric_limits<double>::max();
+  defaultVal.maximum = std::numeric_limits<double>::lowest();
+  defaultVal.count = 0;
+  return mStatisticsMap.value( attribute, defaultVal );
+}
+
+QVariant QgsPointCloudStatistics::statisticOf( const QString &attribute, QgsStatisticalSummary::Statistic statistic ) const
+{
+  if ( !mStatisticsMap.contains( attribute ) )
+    return QVariant();
+  const AttributeStatistics &stats = mStatisticsMap[ attribute ];
+  switch ( statistic )
+  {
+    case QgsStatisticalSummary::Count:
+      return stats.count >= 0 ? QVariant( stats.count ) : QVariant();
+
+    case QgsStatisticalSummary::Min:
+      return stats.minimum;
+
+    case QgsStatisticalSummary::Max:
+      return stats.maximum;
+
+    case QgsStatisticalSummary::Range:
+      return QVariant( stats.maximum - stats.minimum );
+
+    case QgsStatisticalSummary::Mean:
+    case QgsStatisticalSummary::StDev:
+    case QgsStatisticalSummary::CountMissing:
+    case QgsStatisticalSummary::Sum:
+    case QgsStatisticalSummary::Median:
+    case QgsStatisticalSummary::StDevSample:
+    case QgsStatisticalSummary::Minority:
+    case QgsStatisticalSummary::Majority:
+    case QgsStatisticalSummary::Variety:
+    case QgsStatisticalSummary::FirstQuartile:
+    case QgsStatisticalSummary::ThirdQuartile:
+    case QgsStatisticalSummary::InterQuartileRange:
+    case QgsStatisticalSummary::First:
+    case QgsStatisticalSummary::Last:
+    case QgsStatisticalSummary::All:
+      return QVariant();
+  }
+  return QVariant();
+}
+
+QVariantList QgsPointCloudStatistics::classesOf( const QString &attribute ) const
+{
+  if ( !mStatisticsMap.contains( attribute ) )
+    return QVariantList();
+  AttributeStatistics s = mStatisticsMap[ attribute ];
+  QVariantList classes;
+  for ( int c : s.classCount.keys() )
+  {
+    classes.push_back( c );
+  }
+  return classes;
+}
+
+QVariant QgsPointCloudStatistics::classStatisticOf( const QString &attribute, const QVariant &value, QgsStatisticalSummary::Statistic statistic ) const
```

Comment:
What other statistic may we want to calculate in the future for a *specific attribute and value* apart from the count?
I think I'd prefer to have a `::availableClasses( const QString &attribute  )` returning a variant map of available classes and their counts instead.

Reply:
Good idea. And maybe rather than returning QVariantMap, let's return something more specific, e.g. QMap<int, int>

And the same could be applied for minimum() and maximum(), why not just return "double" - and in case the value is not available, we would return NaN.

Reply:
Now that there's `QMap<int, int> availableClasses()`, it seems that `QList<int> classesOf()` is kind of redundant and the naming can be confusing. How about removing it completely?

Reply:
I had the same thought as well :-)